schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AbstractAvroSerDesProtocolHandler.java

```diff
@@ -42,13 +42,13 @@
      */
     public static final String READER_SCHEMA = "reader.schema";
 
-    private final DefaultAvroSerDesHandler defaultAvroSerDesHandler;
+    private final AvroSerDesHandler avroSerDesHandler;
 
     protected final Byte protocolId;
 
-    protected AbstractAvroSerDesProtocolHandler(Byte protocolId) {
+    protected AbstractAvroSerDesProtocolHandler(Byte protocolId, AvroSerDesHandler avroSerDesHandler) {
```

Comment:
There is no code duplication with the current code. `DefaultAvroSerDesHandler` abstracts the default way of handling ser/des payloads and this can be used by any serializer or deserializer like given in exmplaes as `MessageContextSerDesHandler`.